src/test/groovy/com/github/rodionovsasha/commentservice/services/TopicServiceTest.groovy

```diff
@@ -0,0 +1,211 @@
+import org.springframework.data.domain.Sort
+
+class TopicServiceTest extends BaseTest {
+    @Autowired
+    TopicService topicService
+    @Autowired
+    TopicRepository topicRepository
+    final HOMER_ID = 1
+    final TOPIC_ID = 1
+    final NOT_EXISTING_USER_ID = 999
+
+    def "start creates a new topic"() {
+        when:
+        def topic = topicService.start("D'oh!", HOMER_ID)
+
+        then:
+        with(topic) {
+            id instanceof Long
+            title == "D'oh!"
+            !archived
+            date instanceof Date
+        }
+    }
+
+    def "start creates a new topic with owner"() {
+        when:
+        def topic = topicService.start("D'oh!", HOMER_ID)
+
+        then:
+        topic.owner.id == HOMER_ID
+    }
+
+    def "start throws when user is inactive"() {
+        when:
+        topicService.start("D'oh!", 2)
+
+        then:
+        thrown(InactiveUserException)
+    }
+
+    def "start throws when user not found"() {
+        when:
+        topicService.start("D'oh!", NOT_EXISTING_USER_ID)
+
+        then:
+        thrown(UserNotFoundException)
+    }
+
+    def "archive makes a topic archived"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, HOMER_ID)
+
+        then:
+        topicService.getById(TOPIC_ID).archived
+    }
+
+    def "archive throws when user is inactive"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, 2)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        thrown(InactiveUserException)
+    }
+
+    def "archive throws when user not found"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, NOT_EXISTING_USER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        thrown(UserNotFoundException)
+    }
+
+    def "archive throws when user is not topic owner"() {
+        given:
+        def TOPIC_ID = 3
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, HOMER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        def e = thrown(TopicAccessException)
+        e.message == "Non-topic owner with id '1' is trying to archive the topic"
+    }
+
+    def "archive does not change already archived topic"() {
+        given:
+        def TOPIC_ID = 7
+        topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, 1)
+
+        then:
+        topicService.getById(TOPIC_ID).archived
+    }
+
+    def "listForUser returns all topics for user ASC sorted"() {
+        when:
+        def topics = topicService.listForUser(HOMER_ID, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        with(topics) {
```

Comment:
`id == [1, 2, 5, 6, 7]` should completely cover `size() == 5`, so we can simplify: `topics*.id == [1, 2, 5, 6, 7]`